### PR TITLE
Fix typos and grammatical changes

### PR DESCRIPTION
**Pull request by Hemangii**
fixed grammatical errors and typos in 3.10 rst

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@Hemangii

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/whatsnew/3.10.rst [hunk 1]
```diff
@@ -219,7 +219,7 @@ have been incorporated. Some of the most notable ones:
           File "<stdin>", line 1
             {x,y for x,y in zip('abcd', '1234')}
              ^
-        SyntaxError: did you forget parentheses around the comprehension target?
+        SyntaxError: did you forget parenthesis around the comprehension target?
```
Comment (nanjekyejoannah) [inline Doc/whatsnew/3.10.rst]: Am inclined to confirm with @pablogsal before this change, if he didn't actually mean "parentheses" as in plural in this case. It seems to make sense to me as is but again, I dint author it.

Comment (pablogsal) [reply Doc/whatsnew/3.10.rst]: Thanks for checking @nanjekyejoannah !

Indeed, I meant plural. This is also an actual error message so even if we want to change it it should be changed in tended with the source :)

File: Doc/whatsnew/3.10.rst [hunk 2]
```diff
@@ -172,7 +172,7 @@ These improvements are inspired by previous work in the PyPy interpreter.
 :issue:`40176`.)
 
 :exc:`SyntaxError` exceptions raised by the interpreter will now highlight the
-full error range of the expression that consistutes the syntax error itself,
+full error range of the expression that constitues the syntax error itself,
```
Comment (iritkatriel) [inline Doc/whatsnew/3.10.rst]:
```suggestion
full error range of the expression that constitutes the syntax error itself,
```

File: Doc/whatsnew/3.10.rst (hunk 1 above)
Comment (pablogsal) [reply Doc/whatsnew/3.10.rst]:
```suggestion
        SyntaxError: did you forget parentheses around the comprehension target?
```

**Pull request merged**